### ISSUE TITLE
Salsa does not highlight object properties across methods

### ISSUE THREAD

**Issue by egamma** [Microsoft/TypeScript#6664]
_From @alexandrudima on January 27, 2016 12:46_

Testing #2218 

When I hover over `this.age` in the `console.log` I get to see it is a property of type number, but the assignment in line 2 is not highlighted:

```javascript
function Person() {
    this.age = 13;
}
Person.prototype.sayHi = function() {
    console.log('Hi, I\'m ' + this.age);
    console.log('Hi, I\'m ' + this.age);
}

var p1 = new Person();
```

![image](https://cloud.githubusercontent.com/assets/5047891/12613809/5ef74cdc-c4fc-11e5-8d09-2ad0ce47dada.png)

_Copied from original issue: Microsoft/vscode#2438_

**Comment by billti** [comment]
We need to do something here. Seems busted that we highlight a bunch of use sites, but not the declaration.

That said, this is JavaScript, and in a bunch of cases, we are not going to locate references based on type. There are some heuristics we could use, but its always going to be wrong in some cases. We should decide how broken we are happy to live with, if at all.

**Comment by zhengbli** [comment]
This is actually the same with the `find all references` bug, that `getSymbolAtLocation` didn't return the declaration at the `this.^age=` position. This should be fixed also by #6632

**Comment by alexandrudima** [comment]
Awesome! :+1:
